src/run/runner/valgrind/measure.rs

```diff
@@ -42,6 +46,63 @@ lazy_static! {
     };
 }
 
+/// Creates the command that executes the specified command safely.
+///
+/// # Returns
+///
+/// A tuple containing:
+/// 1. The command to execute
+/// 2. The path to the wrapper shell script file
+/// 3. The path to the file containing the exit code
+fn safe_run_cmd(to_execute: &str) -> anyhow::Result<(String, TempPath, TempPath)> {
+    const WRAPPER_SCRIPT: &str = r#"
+        #!/bin/sh
+
+        safe_run()
+        {
+            (eval $1)
+            status=$?
+            echo -n "$status" > "$2"
+        }
+
+        # Args:
+        # 1. Cmd that should be executed
+        # 2. File to write the status code to
+        if [ "$#" -ne 2 ]; then
+            echo "Usage: $0 <cmd> <file>"
+            exit 1
+        fi
+
+        safe_run "$1" "$2"
+        "#;
+
+    // The command is wrapped in a shell script, which executes it in a
+    // subprocess and then writes the exit code to a file. The process will
+    // always exit with status code 0, unless valgrind fails.
+    let script_path = {
+        let rwx = std::fs::Permissions::from_mode(0o777);
+        let mut script_file = tempfile::Builder::new()
+            .suffix(".sh")
+            .permissions(rwx)
+            .tempfile()?;
+        script_file.write_all(WRAPPER_SCRIPT.as_bytes())?;
+
+        // Note: We have to convert the fiel to a path to be able to execute it.
+        // Otherwise this will fail with 'File is busy' error.
+        script_file.into_temp_path()
+    };
+
+    let cmd_status_path = tempfile::NamedTempFile::new().unwrap().into_temp_path();
+    let run_cmd = format!(
+        "{} \"{}\" \"{}\"",
```

Comment:
Defining the command as a string like this seems unsafe. We had many issues with the underlying command already containing quotes and escaped quotes. This quote escaping was handled by `Command` directly. Now, what will happen if we have quotes or escaped quotes in this?